File: 2864.py
class Solution:
    def maximumOddBinaryNumber(self, s: str) -> str:
        return "1" * (s.count('1')-1) + "0" * s.count('0') + "1"

# Topic: String Manipulation / Hash Table / Greedy
# Time Complexity: O(n)
# Space Complexity: O(1)

File: C++/2864.cpp
#include <string>
#include <bitset>

class Solution {
public:
    string maximumOddBinaryNumber(string s) {
        return string(count(s.begin(), s.end(), '1')-1, '1') + string(s.length()-count(s.begin(), s.end(), '1'), '0') + '1';

    }
};

// Time Complexity: O(n)
// Space Complexity: O(1)